itests/kit/node_unmanaged.go

```diff
@@ -24,3 +60,730 @@ type TestUnmanagedMiner struct {
+	_, err = pieceFileA.Seek(0, io.SeekStart)
+	require.NoError(tm.t, err)
+
+	unsealedSectorFile := requireTempFile(tm.t, bytes.NewReader([]byte{}), 0)
+	defer func() {
+		_ = unsealedSectorFile.Close()
+	}()
+
+	// Write the piece to the staged sector file without alignment
+	writtenBytes, pieceCID, err := ffi.WriteWithoutAlignment(pt, pieceFileA, unpaddedPieceSize, unsealedSectorFile)
+	require.NoError(tm.t, err)
+	require.EqualValues(tm.t, unpaddedPieceSize, writtenBytes)
+	require.True(tm.t, pieceCID.Equals(pieceCIDA))
+
+	// Create a struct for the piece info
+	publicPieces := []abi.PieceInfo{{
+		Size:     paddedPieceSize,
+		PieceCID: pieceCIDA,
+	}}
+
+	// Create a temporary file for the sealed sector
+	sealedSectorFile := requireTempFile(tm.t, bytes.NewReader([]byte{}), 0)
+	defer func() {
+		_ = sealedSectorFile.Close()
+	}()
+
+	// Update paths for the sector
+	tm.sealedSectorPaths[sectorNumber] = sealedSectorFile.Name()
+	tm.unsealedSectorPaths[sectorNumber] = unsealedSectorFile.Name()
+	tm.cacheDirPaths[sectorNumber] = filepath.Join(tm.cacheDir, fmt.Sprintf("%d", sectorNumber))
+
+	// Ensure the cache directory exists
+	_ = os.Mkdir(tm.cacheDirPaths[sectorNumber], 0755)
+
+	return publicPieces
+}
+
+func (tm *TestUnmanagedMiner) makeAndSaveCCSector(_ context.Context, sectorNumber abi.SectorNumber) {
+	requirements := require.New(tm.t)
+
+	// Create cache directory
+	cacheDirPath := filepath.Join(tm.cacheDir, fmt.Sprintf("%d", sectorNumber))
+	requirements.NoError(os.Mkdir(cacheDirPath, 0755))
+	tm.t.Logf("MinerB: Sector %d: created cache directory at %s", sectorNumber, cacheDirPath)
+
+	// Define paths for unsealed and sealed sectors
+	unsealedSectorPath := filepath.Join(tm.unsealedSectorDir, fmt.Sprintf("%d", sectorNumber))
+	sealedSectorPath := filepath.Join(tm.sealedSectorDir, fmt.Sprintf("%d", sectorNumber))
+	unsealedSize := abi.PaddedPieceSize(sectorSize).Unpadded()
+
+	// Write unsealed sector file
+	requirements.NoError(os.WriteFile(unsealedSectorPath, make([]byte, unsealedSize), 0644))
+	tm.t.Logf("MinerB: Sector %d: wrote unsealed CC sector to %s", sectorNumber, unsealedSectorPath)
+
+	// Write sealed sector file
+	requirements.NoError(os.WriteFile(sealedSectorPath, make([]byte, sectorSize), 0644))
+	tm.t.Logf("MinerB: Sector %d: wrote sealed CC sector to %s", sectorNumber, sealedSectorPath)
+
+	// Update paths in the struct
+	tm.unsealedSectorPaths[sectorNumber] = unsealedSectorPath
+	tm.sealedSectorPaths[sectorNumber] = sealedSectorPath
+	tm.cacheDirPaths[sectorNumber] = cacheDirPath
+}
+
+func (tm *TestUnmanagedMiner) OnboardSectorWithPiecesAndRealProofs(ctx context.Context, proofType abi.RegisteredSealProof) (abi.SectorNumber, chan WindowPostResp) {
+	req := require.New(tm.t)
+	sectorNumber := tm.currentSectorNum
+	tm.currentSectorNum++
+
+	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
```

Comment:
this function's docs all talk about "CC" but there's a piece in here, right?

Reply:
Fixed.